### PR TITLE
feature: adds infinity scroll to flat list

### PR DESCRIPTION
**Pull request by tkumor3**
https://github.com/user-attachments/assets/6a1b5693-fb0a-4ccf-8796-0110ce3af96a

### CODE REVIEW

File: src/hooks/usePokemons.ts [hunk 1]
```diff
@@ -32,8 +45,30 @@ const usePokemons = () => {
       };
     }) ?? [];
 
+  const handleLoadMore = async () => {
```
Comment (mironiasty) [inline src/hooks/usePokemons.ts:48]: in general, if you return function from hook -> it is always good idea to use `useCallback`

File: src/App.tsx [hunk 2]
```diff
@@ -32,7 +32,27 @@ const link = from([
...
 const client = new ApolloClient({
   link,
-  cache: new InMemoryCache(),
+  cache: new InMemoryCache({
+    typePolicies: {
+      Query: {
+        fields: {
+          pokemon_v2_pokemon: {
+            keyArgs: false,
+            //TODO: remove any after merge generates types
```
Comment (mironiasty) [inline src/App.tsx:41]: 👀

File: src/hooks/usePokemons.ts [hunk 3]
```diff
@@ -32,8 +45,30 @@ const usePokemons = () => {
...
+    if (loadedAll) {
+      return;
+    }
+    setLoadingMore(true);
+    try {
+      const result = await fetchMore({
```
Comment (mironiasty) [inline src/hooks/usePokemons.ts:54]: what will happen if this call will fail?

Comment (tkumor3) [reply src/hooks/usePokemons.ts:54]: nothing IMO in this case seems to be ok

File: src/App.tsx (hunk 2 above)
Comment (tkumor3) [reply src/App.tsx]: It's Apollo type, I have no idea how to adds type here

**Review by mironiasty — APPROVED** [review]

**Pull request merged**